lib/diagnostic.js

```diff
@@ -9,15 +9,40 @@
 // Write a callback for reduce that selects
 // the minimum value in the array
 const min = function min() {
+    let args;
+    if (arguments[0] instanceof Array) {
+        args = arguments[0];
+    } else {
+        args = arguments;
+    }
+    let answer = args[0];
 
+    for(let i = 1; i < args.length; i++) {
+        if(answer > args[i]) {
+            answer += args[i];
+        }
+        return answer;
+    }
 };
 
+//array.reduce(function())
+
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
-const sum = function sum() {
+// const sum = function sum(array) {
+    let answer = array[0];
 
+    for(let i = 1; i < array.length; i++) {
+        answer += array[i];
+    }
+    return answer;
 };
+let array = [2,4,6];
+array.reduce(function(a, b) {
+    return a + b;
+});
```

Comment:
I think you sort of arrived at the right answer eventually.
```
const sum = function(a,b) {
  return a + b;
}
```
When you run `[2,4,6].reduce(sum)` you should get a result of 12